sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineRunner.java

```diff
@@ -420,6 +426,37 @@ public static DataflowPipelineRunner fromOptions(PipelineOptions options) {
     return super.apply(new AssignWindows<>(transform), input);
   }
 
+  @Nullable
+  private void maybeRegisterDebuggee(DataflowPipelineOptions options, String uniquifier) {
```

Comment:
I thought about putting this somewhere else, but we'd end up with a "DebuggerUtility" that was essentially this code. Given the high coupling with when it needs to be called (eg., it needs to be called before translating the job, it needs to have the uniquifier, etc) I don't think that getting these ~45 lines into a separate module helps much (eg., we're unlikely to reuse it, etc.).

Open to thoughts if you had a specific suggestion for improvement.